pkg/endpoints/pipeline_test.go

```diff
@@ -1167,3 +1167,137 @@ func TestPOSTCreatePipelineRuns(t *testing.T) {
 		}
 	}
 }
+
+func createPipelineForTesting(name, namespace string, server *httptest.Server, r *Resource, t *testing.T) *v1alpha1.Pipeline {
+	// Pipelines referenced by PipelineRuns
+	pipeline := &v1alpha1.Pipeline{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.PipelineSpec{},
+	}
+	_, err := r.PipelineClient.TektonV1alpha1().Pipelines(namespace).Create(pipeline)
+	if err != nil {
+		t.Fatalf("Error creating Pipeline '%s': %v", pipeline.Name, err)
+	}
+	return pipeline
+}
+
+func createPipelineRunForTesting(pipelineNameToUse, namespace string, server *httptest.Server, r *Resource, t *testing.T) *v1alpha1.PipelineRun {
+	manualPipelineRun := ManualPipelineRun{
+		PIPELINENAME:      pipelineNameToUse,
+		IMAGERESOURCENAME: "image-source-" + pipelineNameToUse,
+		GITRESOURCENAME:   "git-source - " + pipelineNameToUse,
+		GITCOMMIT:         "12345",
+		REPONAME:          "testreponame",
+		REPOURL:           "http://github.com/testorg/testrepo",
+	}
+
+	jsonMarshalledBytes, err := json.Marshal(&manualPipelineRun)
+	if err != nil {
+		t.Fatalf("Error marshalling manualPipelineRun: %v", err)
+	}
+
+	httpRequestBody := bytes.NewReader(jsonMarshalledBytes)
+	//
+	httpReq := testutils.DummyHTTPRequest("POST", fmt.Sprintf("%s/v1/namespaces/%s/pipelineruns", server.URL, namespace), httpRequestBody)
+	response, err := http.DefaultClient.Do(httpReq)
+
+	if err != nil {
+		t.Fatalf("Error getting create PipelineRun response for PipelineRun '%s': %v", pipelineNameToUse, err)
+	}
+	expectedStatusCode := 201
+	if response.StatusCode != expectedStatusCode {
+		t.Fatalf("Response code %d did not equal expected %d", response.StatusCode, expectedStatusCode)
+	}
+
+	foundPipelineRuns, err := r.PipelineClient.TektonV1alpha1().PipelineRuns(namespace).List(metav1.ListOptions{})
+	if len(foundPipelineRuns.Items) != 1 {
+		t.Errorf("Didn't find the PipelineRun to rebuild")
+	}
+	return &foundPipelineRuns.Items[0]
+}
+
+// Create a PipelineRun then rebuild it using the "from name" API.
+// Ensure it's identical apart from the name and resource version
+
+func TestRebuildWithExistingRunFromName(t *testing.T) {
+	server, r, namespace := testutils.DummyServer()
+	testPipelineNameToRun := "testPipeline1"
+
+	pipeline := createPipelineForTesting(testPipelineNameToRun, namespace, server, r, t)
+	pipelineRun := createPipelineRunForTesting(testPipelineNameToRun, namespace, server, r, t)
+
+	// Check we've got a new PipelineRun and its the same apart from name and resource version
```

Comment:
this comment confused me, isn't this supposed to be on the second creation only?